metadata-ingestion/src/datahub/ingestion/source/unity/usage.py

```diff
@@ -0,0 +1,217 @@
+                for source_table in table_info.source_tables:
+                    self.usage_aggregator.aggregate_event(
+                        resource=source_table,
+                        start_time=query.start_time,
+                        query=query.query_text,
+                        user=query.user_name,
+                        fields=[],
+                    )
+
+        for wu in self.usage_aggregator.generate_workunits(
+            resource_urn_builder=self.table_urn_builder,
+            user_urn_builder=self.user_urn_builder,
+        ):
+            self.report.num_usage_workunits_emitted += 1
+            yield wu
+
+    def _generate_operation_workunit(
+        self, query: Query, table_info: QueryTableInfo
+    ) -> Iterable[MetadataWorkUnit]:
+        if query.statement_type not in OPERATION_STATEMENT_TYPES:
+            return None
+
+        # Not sure about behavior when there are multiple target tables. This is a best attempt.
+        for target_table in table_info.target_tables:
+            operation_aspect = OperationClass(
+                timestampMillis=int(time.time() * 1000),
+                lastUpdatedTimestamp=int(query.end_time.timestamp() * 1000),
+                actor=self.user_urn_builder(query.user_name),
+                operationType=OPERATION_STATEMENT_TYPES[query.statement_type],
+                affectedDatasets=[
+                    self.table_urn_builder(table) for table in table_info.source_tables
+                ],
+            )
+            self.report.num_operational_stats_workunits_emitted += 1
+            yield MetadataChangeProposalWrapper(
+                entityUrn=self.table_urn_builder(target_table), aspect=operation_aspect
+            ).as_workunit()
+
+    def _get_queries(self) -> Iterable[Query]:
+        try:
+            yield from self.proxy.query_history(
+                self.config.start_time, self.config.end_time
+            )
+        except Exception as e:
+            logger.warning("Error getting queries", exc_info=True)
+            self.report.report_warning("get-queries", str(e))
+
+    def _parse_query(
+        self, query: Query, table_map: TableMap
+    ) -> Optional[QueryTableInfo]:
+        table_info = self._parse_query_via_lineage_runner(query.query_text)
+        if table_info is None and query.statement_type == StatementType.SELECT:
+            table_info = self._parse_query_via_spark_sql_plan(query.query_text)
+
+        if table_info is None:
+            self.report.num_queries_dropped_parse_failure += 1
+            return None
+        else:
+            return QueryTableInfo(
+                source_tables=self._resolve_tables(table_info.source_tables, table_map),
+                target_tables=self._resolve_tables(table_info.target_tables, table_map),
+            )
+
+    def _parse_query_via_lineage_runner(self, query: str) -> Optional[StringTableInfo]:
+        try:
+            runner = LineageRunner(query)
```

Comment:
not a huge fan of the fact that this LineageRunner stuff is copy-pasted across our codebase

Ideally we'd go through a unified parser interface, where you can select the underlying parser(s) to use with an enum or something

Reply:
don't need to fix it here though

Reply:
We have a our `SqlLineageSQLParser` which creates `SqlLineageSQLParserImpl` which eventually calls `LineageAnalyzer().analyze`, but there was a lot of logic in there which didn't seem necessary. I thought overall this was simpler. Agree that we should standardize this as much as possible, but I think it'll take a bit of work